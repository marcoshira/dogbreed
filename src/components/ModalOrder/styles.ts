/* istanbul ignore file */
import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    img {
      width: ${theme.frameSizes.medium};
      height: ${theme.frameSizes.medium};
    }

    @media ${theme.media.lteOrEqMedium} {
      width: ${theme.frameSizes.largeMedium};
    }

    @media ${theme.media.lteOrEqSmall} {
      width: 100vw;
    }

    @media ${theme.media.lteSmall} {
      width: 100vw;
      height: 55vh;
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    z-index: 6;
    position: fixed;
    top: 2rem;
    right: 2rem;
    width: 3rem;
    height: 3rem;
    background: none;
    color: ${theme.colors.white};
    border: none;
    display: flex;
    align-items: center;
    justify-content: center;
    transition: ${theme.transitions.fastest};
    cursor: pointer;
    > svg {
      width: 3rem;
      height: 3rem;
    }

    &:hover {
      transform: scale(1.05);
      color: ${theme.colors.black};
      /* background: ${theme.colors.white}; */
    }
  `}
`;
