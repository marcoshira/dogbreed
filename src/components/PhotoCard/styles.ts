import styled, { css } from 'styled-components';

export const Wrapper = styled.a`
  ${({ theme }) => css`
    display: inline-block;
    margin: 10px;
    cursor: pointer;
    overflow: hidden;
    height: 260px;
    width: 260px;
    border-radius: 5%;

    img {
      height: 260px;
      width: 260px;
      transition: all 300ms ease-in-out;

      object-fit: cover;

      &:hover {
        transform: scale(1.2) rotate(10deg);
      }
    }
  `}
`;
