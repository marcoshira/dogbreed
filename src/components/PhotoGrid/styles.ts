import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(260px, 1fr));
    gap: 0;
    @media ${theme.media.lteSmall} {
      grid-template-columns: 1fr;
    }
  `}
`;
