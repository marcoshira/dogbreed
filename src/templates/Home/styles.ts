import styled from 'styled-components';
import { Wrapper as Container } from '../../components/Wrapper/styles';

export const Wrapper = styled.div`
  ${Container} {
    max-width: 90rem;
  }
`;
