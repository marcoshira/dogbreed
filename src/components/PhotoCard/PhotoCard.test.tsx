import { fireEvent, screen, waitFor } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { PhotoCard, PhotoCardProps } from '.';

const func = jest.fn();

const props: PhotoCardProps = {
  srcImg: 'img.png',
  onClick: func,
};

describe('<PhotoCard />', () => {
  it('should render', () => {
    renderTheme(<PhotoCard {...props} />);

    expect(screen.getByRole('img')).toHaveAttribute('src', 'img.png');
  });
  it('should call function', async () => {
    renderTheme(<PhotoCard {...props} />);
    fireEvent.click(screen.getByRole('img'));
    await waitFor(() => {
      expect(func).toHaveBeenCalled();
    });
  });
  it('should match snapshot', () => {
    const { container } = renderTheme(<PhotoCard {...props} />);

    expect(container).toMatchSnapshot();
  });
});
