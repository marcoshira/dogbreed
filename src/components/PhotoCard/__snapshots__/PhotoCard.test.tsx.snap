// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhotoCard /> should match snapshot 1`] = `
.c0 {
  display: inline-block;
  margin: 10px;
  cursor: pointer;
  overflow: hidden;
  height: 260px;
  width: 260px;
  border-radius: 5%;
}

.c0 img {
  height: 260px;
  width: 260px;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  object-fit: cover;
}

.c0 img:hover {
  -webkit-transform: scale(1.2) rotate(10deg);
  -ms-transform: scale(1.2) rotate(10deg);
  transform: scale(1.2) rotate(10deg);
}

<div>
  <a
    class="c0"
  >
    <img
      src="img.png"
    />
  </a>
</div>
`;
